src/target_set.c

```diff
@@ -28,6 +28,28 @@
 
 #include "common.h"
 #include "filter_chain.h"
+#include "meta_data.h"
+#include "utils_subst.h"
+
+struct ts_key_list_s
```

Comment:
I've tried it out, but the cost of using the `strarray...` functions is having two data members that have to be passed around, which is somewhat ugly. If you don't mind, I'll stick with the current approach (why don't we have convenience functions for linked lists again?). If you still think I should be using `strarray...` here, please let me know and I'll change it.